src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,171 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import {
+  Button, Form, Hyperlink, ModalLayer, Spinner, TransitionReplace,
+} from '@edx/paragon';
+import { Formik } from 'formik';
+import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import * as Yup from 'yup';
+import { RequestStatus } from '../../data/constants';
+import FormSwitchGroup from '../../generic/FormSwitchGroup';
+import { useModel } from '../../generic/model-store';
+import StatusBadge from '../../generic/status-badge/StatusBadge';
+import { getLoadingStatus } from '../data/selectors';
+import { enableDisableApp } from '../data/thunks';
+import AppConfigFormDivider from '../discussions/app-config-form/apps/shared/AppConfigFormDivider';
+import { PagesAndResourcesContext } from '../PagesAndResourcesProvider';
+import messages from './messages';
+
+function AppSettingsForm({ formikProps, children }) {
+  return children && (
+    <TransitionReplace>
+      {formikProps.values.enabled
+        ? (
+          <React.Fragment key="app-enabled">
+            {children(formikProps)}
+          </React.Fragment>
+        ) : (
+          <React.Fragment key="app-disabled" />
+        )}
+    </TransitionReplace>
+  );
+}
+
+AppSettingsForm.propTypes = {
+  formikProps: PropTypes.shape({
+    values: PropTypes.shape({ enabled: PropTypes.bool.isRequired }),
+  }).isRequired,
+  children: PropTypes.func,
+};
+
+AppSettingsForm.defaultProps = {
+  children: null,
+};
+
+function AppSettingsModal({
+  intl,
+  appId,
+  title,
+  children,
+  initialValues,
+  validationSchema,
+  onClose,
+  onSettingsSave,
+  enableAppLabel,
+  enableAppHelp,
+  learnMoreURL,
+  learnMoreText,
+}) {
+  const { courseId } = useContext(PagesAndResourcesContext);
+  const loadingStatus = useSelector(getLoadingStatus);
+  const appInfo = useModel('courseApps', appId);
+  const dispatch = useDispatch();
+  const handleFormSubmit = (values) => {
+    // If the app's enabled/disabled loadingStatus has changed, set that first.
+    if (appInfo.enabled !== values.enabled) {
+      dispatch(enableDisableApp(courseId, appInfo.id, values.enabled, true));
+    }
+    // Call the submit handler for the settings component to save its settings
+    if (onSettingsSave) {
+      onSettingsSave();
+    }
+  };
+  const learnMoreLink = (
+    <Hyperlink destination={learnMoreURL} target="_blank">{learnMoreText}</Hyperlink>
```

Comment:
for security purposes please also add `rel="noopener noreferrer"` in the element properties.